openedx/core/djangoapps/util/helpers.py

```diff
@@ -0,0 +1,48 @@
+"""
+Common helpers methods for django apps.
+"""
+
+import logging
+
+from django.core.exceptions import ImproperlyConfigured
+from provider.oauth2.models import AccessToken, Client
+from provider.utils import now
+
+
+log = logging.getLogger(__name__)
+
+
+def get_id_token(user, client_name):
+    """Generates a JWT ID-Token, using or creating user's OAuth access token.
+
+    Arguments:
+        user (User Object): User for which we need to get JWT ID-Token
+        client_name (unicode): Name of the OAuth2 Client
+
+    Returns:
+        String containing the signed JWT value or raise the exception
+        'ImproperlyConfigured'
+    """
+    # TODO: there's a circular import problem somewhere which is why we do the oidc import inside of this function.
```

Comment:
@jimabramson `ally` tests still failing due to circular import. The full traceback is here: https://gist.github.com/zubair-arbi/bec778d5d3e7c0ca79ac